framework/src/interfacekernels/InterfaceKernel.C

```diff
@@ -216,7 +220,9 @@ InterfaceKernelTempl<T>::computeResidual()
   computeElemNeighResidual(Moose::Element);
 
   // Compute the residual for the neighbor
-  computeElemNeighResidual(Moose::Neighbor);
+  // This also prevents computing a residual if the neighbor variable is auxiliary
+  if (_same_system)
+    computeElemNeighResidual(Moose::Neighbor);
```

Comment:
So if the neighbor variable was from a nonlinear system that is different than the element variable, would the neighbor residual ever get computed?

Reply:
does your code work if the aux variable is the element var vs. the neighbor var?

Reply:
No there s no way. You can't set the variable parameter to an aux variable, you ll get a "did you mean to use ..." from the MooseVariableInterface

Reply:
Hmm, I did not think through all the repercussions before sending you down this goose chase. My inclination at this point is to ban mixed systems since it seems odd to support mixed nl-aux only "by luck", e.g. the user happens to assign the aux variable to the neighbor side. I suppose if it was well documented in the interface kernel that you can use an aux variable, but only on the neighbor side, then we could move ahead. What do you think?

Reply:
As you want. I'd merge this since it fixes the seg fault the user got without breaking anything. I can also work on making multi system work. I think I singled out the places to add code to already at this point.

Reply:
We can merge, but can you add something in `framework/doc/content/syntax/InterfaceKernels/index.md`?

Reply:
done